packages/frontend/src/components/userActious/gitcoin/CheckingGitcoinScore.tsx

```diff
@@ -31,7 +37,34 @@ const gitcoinScoreSteps = [
   },
 ]
 
-export const CheckGitcoinScore = () => {
+interface CheckGitcoinScoreProps {
+  setGitcoinCredentials: (credentials: GetScoreResponseSuccess) => void
+}
+
+export const CheckGitcoinScore = ({ setGitcoinCredentials }: CheckGitcoinScoreProps) => {
+  const { address } = useAccount()
+  const chainId = useChainId()
+  const { mutate, requestSettled } = useSendForScoring()
+
+  const { data } = useQuery({
+    queryKey: ['gitcoinScore', address, chainId],
+    queryFn: () => getGitcoinScore(address as Hex, chainId),
+    enabled: !!(requestSettled && address),
+    retry: true,
+    retryDelay: 5000,
+  })
+  const step = requestSettled ? 2 : 1
+
+  useEffect(() => {
+    mutate()
```

Comment:
I believe that we can have some optional onChange prop in step that can be invoked when we change step

Reply:
I have refactored this logic so there is no more useEffect